### ISSUE TITLE
Give the training loop a schedule, warmup, decoupled weight decay, and clipping

### ISSUE THREAD

**Issue by mmmfrieddough** [mmmfrieddough/anthro-chess#493]
## Goal

Give the training loop the optimization surface the scaling program assumes it
has, and which it does not have today.

## What Is Missing

`src/anthro_chess/training/config.py` carries a single `learning_rate` float, and
`runner.py` builds a plain `torch.optim.Adam`. There is:

- **no schedule of any kind** — no warmup, no decay, no cooldown;
- **no weight decay**, decoupled or otherwise, because the optimizer is `Adam`
  rather than `AdamW`;
- **no configurable second-moment decay;**
- **no gradient clipping**, anywhere under `src/anthro_chess/training/`.

Every one of those is a setting `docs/scaling.md` treats as scale-coupled and
that later issues are written against. They cannot be recorded as rules before
they can be set at all.

## Why It Blocks The Vehicle

The vehicle's identity digest covers its training configuration, so a schedule
added after `#487` freezes it changes the digest and invalidates every comparison
read against it —
`docs/decisions/0065-a-frozen-ablation-vehicle-is-the-base-a-seed-floor-can-live-on.md`
is explicit that this is the failure the pin exists to make loud. The surface has
to exist before the freeze, not after it.

`#486` decides which schedule family; this issue is what makes that decision
expressible.

Gradient clipping was pulled forward from `#499` for the same reason and one
further one. It is inside the training identity, so adding it after the freeze
carries the same cost as adding a schedule after the freeze. And the vehicle is
the instrument every Milestone 5 comparison is read against: a base prone to loss
spikes has its seed dispersion inflated by spike luck, which widens the floor and
blunts every candidate reading taken against it. The schedule family settled in
`#486` sharpens that — a constant trunk holds the peak rate for the whole run
rather than decaying away from it, so there is no low-rate second half where a
freak batch is harmless.

The rest of `#499`'s package stays where it is. Only clipping moves, because only
clipping has the vehicle waiting on it.

## Acceptance Criteria

- The chosen schedule family from `#486` is expressible in checked-in
  configuration and is what the runner uses.
- Warmup is configured as a rule against scale rather than as a fixed step count.
  A fixed warmup is one of the four confounds `docs/research.md` (Resolving
  Discrepancies In Compute-Optimal Scaling) isolates, and it was worth a large
  share of a published exponent error.
- Weight decay is available and decoupled, so the timescale it implies is the
  quantity a later issue can hold fixed. Whether the project uses a nonzero value
  is a separate question this issue does not settle; what it settles is that the
  knob exists.
- The second-moment decay is configurable, since it is a timescale in positions
  rather than a scale-free constant and moves when the batch does.
- Where a cooldown is adopted, branching one from a mid-run checkpoint is a
  documented command. `resume_from` already exists; this issue establishes that
  branching produces a correctly-shaped cooldown rather than a resumed trunk.
- Cooldown length and shape are set as fractions of the run rather than as step
  counts, so a horizon change does not silently reshape them.
- Existing configurations either keep their current behavior or are migrated
  deliberately, and the issue says which. A silent change to what existing
  configs train is a change to every reading taken under them.
- Unit coverage for schedule shape at the boundaries — step zero, end of warmup,
  start of cooldown, final step — since an off-by-one in a schedule is invisible
  in a loss curve.
- Gradient clipping exists, by global norm, applied to the accumulated gradient
  before the optimizer step. `StepHealthMonitor.observe_gradients` already
  computes that norm on every step through `torch.nn.utils.get_total_norm`, and
  the pinned Torch exposes `clip_grads_with_norm_(parameters, max_norm,
  total_norm)`, so the reduction this needs is one the runner already pays for.
- Its threshold is set from gradient norms a real run recorded rather than copied
  from a framework default, and it is loose enough to be inactive in ordinary
  operation. A threshold that fires routinely is a learning-rate cap under
  another name, and it would confound the peak-rate rule `#489` owes.
- The clip rate is recorded beside the gradient norm. Without it, insurance that
  never fires and a silent cap on every step look identical from the loss curve,
  which is the failure `#499` names for the rest of the package.

## Earning The Surface

This adds configuration options, so per `docs/design-principles.md` each one is
justified by a decision that needs it: the schedule family by `#486`, the weight
decay and second-moment knobs by `#489`, the clip threshold by the freeze at
`#487`. An option no later issue reads is removed rather than shipped unused.

## Model-Change Reading

This changes what a training run learns, so it owes the reading
`docs/issue-workflow.md` (Showing A Model Change Improved Something) describes.
The vehicle does not exist yet, so the control is the canonical line's prior
checkpoint under
`docs/decisions/0063-the-full-sweep-decides-a-change-and-the-canonical-line-is-its-byproduct.md`,
and the comparison carries no seed floor. Say so rather than reporting a
`cleared` verdict as a result.

## References

- `docs/scaling.md` — which of these are scale-coupled and how hard
- `docs/research.md` (Scaling And Capacity)
- `docs/training-and-runtime.md`
- `#486` — the decision this makes expressible

**Comment by mmmfrieddough** [comment]
The schedule family this issue makes expressible is settled in #501 — \`docs/decisions/0067-a-horizon-is-a-branch-not-a-restart.md\`, with the rule in \`docs/scaling.md\`. Three constraints in it are narrower than this issue's criteria and are worth reading before implementing:

- Warmup is denominated in **training data** (the sequences the loader delivers, converted to steps by the declared batch and accumulation), not in steps and not as a fraction of the run. A horizon-relative warmup would give each branch a different prefix and there would be no shared trunk to branch from.
- The rule is recorded with the range it is valid over — warmup under roughly a tenth of the run — and a configuration outside it is refused rather than quietly reshaped.
- A branch **restores no schedule state**. The rate is recomputed from the resumed step against the resuming run's own horizon; a restored \`scheduler_state\` would carry the trunk's horizon into the branch and cool at the wrong boundary with nothing reporting it. \`docs/training-and-runtime.md\` now states that, and it is the case worth a boundary test.

**Comment by mmmfrieddough** [comment]
Claimed for implementation by Claude Code (Opus 5).

Session: e0e61800-3f0d-49f7-9793-e63d95c779dd

**Comment by mmmfrieddough** [comment]
Pre-registering the claim the model-change reading tests, before either arm's comparison exists.

**The change under test.** One arm trains under the schedule family `#501` settled (warmup over a declared quantity of training data, a constant trunk at the peak rate, a square-root cooldown over the final fraction of the horizon) with the global-norm ceiling active. The control arm holds the peak rate for the whole run with the ceiling set loose enough to be inert, which is what every configuration in the repository does today. Same corpus, seed, horizon, batch, device, and model.

**What carries it.** `held_out.move_loss` over the frozen pool, read on each arm's final checkpoint with `anthro eval run`. The cooled arm is expected lower; `held_out.top1_accuracy` higher is the secondary, and a split between the two is a finding rather than a pass. `training_health.clip_rate` on the treatment arm is expected at zero, which is what says the ceiling is insurance rather than a rate cap.

**What it will not carry.** One seed and no vehicle, so no seed floor applies and a `cleared` verdict on the evaluation floor alone would say only that the delta survived a different draw of evaluation units.

Also recorded before the arms run, since it decides how the reading is reported: at neutral settings the change is bit-inert. Six steps on the sample corpus under strict determinism on CPU reach the same final parameter digest (`69fe1b6a...`) before and after, so `AdamW` at zero weight decay reproduces `Adam` exactly and a run that declares no warmup and no cooldown trains what it trained before.

**Comment by mmmfrieddough** [comment]
The held-out reading this issue registered a claim for, and that `#514` shipped without, has now been taken. `#517`, `#518` and `#520` removed the costs that were stopping it, so both arms were read over the whole frozen pool rather than a subsample.

**Both arms, canonical view, 211,475 games and 14,161,038 scored positions each.**

| metric | control | cooled | delta |
| --- | --- | --- | --- |
| `move_loss` | 2.103347 | 2.043862 | **-0.059485** |
| `legal_move_loss` | 1.992073 | 1.940565 | -0.051508 |
| `accuracy.top1` | 0.402636 | 0.414057 | **+0.011421** |
| `accuracy.top3` | 0.672275 | 0.684562 | +0.012286 |
| `accuracy.top5` | 0.785004 | 0.795862 | +0.010858 |
| `legality.mask_penalty` | 0.111274 | 0.103297 | -0.007977 |
| `legality.top1_illegal_rate` | 0.037245 | 0.036243 | -0.001002 |
| `legality.legal_mass` | 0.909911 | 0.915720 | +0.005809 |

**The registered claim holds, in the direction it named.** Move loss is lower and top-1 accuracy is higher, so there is no split to write up. Taking the worst whole-pool dispersion in either record, 0.001214, and combining two readings at that worst case gives a floor of 0.001716: the move-loss delta is 34.7 times it and the top-1 delta 6.7 times it. Nothing marginal here.

**What it does not carry, unchanged from what the pull request said.** One seed and no vehicle, so no seed floor applies and this is unqualified on training-seed noise. Both arms sit at the same untuned 1e-3 peak. The delta is attributable to the schedule alone rather than to the schedule plus clipping, because the ceiling never fired in either arm: the largest interval norm was 9.57 in the control and 7.42 in the cooled arm against a ceiling of 10, and `clip_rate` is 0.0 throughout both runs.

**Reproduction.** The checkpoints are retained as `schedule-arm-control` and `schedule-arm-cooled` beneath this machine's run root, both at `step-00008000.pt`:

```console
uv run anthro eval run \
  --config configs/evaluation/lichess-univ-2018-01-2021-06-checkpoint.toml \
  --no-record --format json --set 'model.run_path="schedule-arm-control"'
```

Each reading took about 25 minutes and peaked at 9.7 GB, with both running at once on separate accelerators. That is also the end-to-end canonical run `#518` and `#520` each noted they had not done: their extrapolation of roughly 22 minutes holds, and the memory sits a little above the 5 to 8 GB `#518` composed from separately measured parts.
